tests/unit/cypher_main_visitor.cpp

```diff
@@ -3803,6 +3842,44 @@ TEST_P(CypherMainVisitorTest, CreateKafkaStream) {
   for (const auto consumer_group : consumer_groups) {
     EXPECT_NO_FATAL_FAILURE(check_consumer_group(consumer_group));
   }
+
+  auto check_config_map = [&](const std::unordered_map<std::string, std::string> &config_map) {
+    const std::string kTopicName{"topic1"};
+
+    const auto map_as_str = std::invoke([&config_map] {
+      std::stringstream buffer;
+      buffer << '{';
+      if (!config_map.empty()) {
+        const auto &first_item = *config_map.begin();
+        buffer << fmt::format("'{}': '{}'", first_item.first, first_item.second);
+        for (auto it = ++config_map.begin(); it != config_map.end(); ++it) {
+          buffer << fmt::format(", '{}': '{}'", it->first, it->second);
+        }
+      }
+      buffer << '}';
+      return std::move(buffer).str();
+    });
+
+    ValidateCreateKafkaStreamQuery(ast_generator,
+                                   fmt::format("CREATE KAFKA STREAM {} TOPICS {} TRANSFORM {} CONFIGS {}", kStreamName,
+                                               kTopicName, kTransformName, map_as_str),
+                                   kStreamName, {kTopicName}, kTransformName, "", std::nullopt, std::nullopt, {},
+                                   config_map, {});
+
+    ValidateCreateKafkaStreamQuery(ast_generator,
+                                   fmt::format("CREATE KAFKA STREAM {} TOPICS {} TRANSFORM {} CREDENTIALS {}",
+                                               kStreamName, kTopicName, kTransformName, map_as_str),
+                                   kStreamName, {kTopicName}, kTransformName, "", std::nullopt, std::nullopt, {}, {},
+                                   config_map);
+  };
+
+  const std::array<std::unordered_map<std::string, std::string>, 3> config_maps = {
```

Comment:
```suggestion
  const std::array config_maps = {
```
Or you don't define type for each element, no need for both

Reply:
Removing the types from each element makes the second (and probably the third) one fail to compile (with error `error: excess elements in struct initializer`). I don't know the exact reason, but as it is not critical, I removed the template types of the array.